Assets/Scripts/CameraController.cs

```diff
@@ -5,57 +5,89 @@
 public class CameraController : MonoBehaviour
 {
     [SerializeField] private float minZoom = 5f;
-    
     [SerializeField] private float maxZoom = 15f;
-
-    [Tooltip("The scroll damp time in seconds")]
     [SerializeField] private float scrollDampTime = 0.5f;
+    [SerializeField] private float edgeMoveDampTime = 0.5f;
+    [SerializeField] private float scrollSensitivity = 0.1f;
+    [SerializeField] private float edgeCamMoveThreshold = 0.1f;
+    [SerializeField] private float maxCameraMoveSpeed = 0.1f;
+    [SerializeField] private float edgeMoveAcceleration = 0.1f;
+    [SerializeField] private bool invertScrolling = true;
 
-    [SerializeField] private float scrollSensitivity = 1f;
+    Dampable edgeMoveXDampable;
+    Dampable edgeMoveYDampable;
+    Dampable scrollMoveDampable;
 
-    [SerializeField] private float scrollThreshold = 0.1f;
+    private float halfWidth;
+    private float halfHeight;
 
-    private float scrollVelocity = 0f;
-    private float lastScrollVelocity = 0f;
-    private float scrollDampTimer;
-    private Vector2 lastMouseViewportPos = new Vector2(0.5f, 0.5f);
-    private Vector2 lastMouseWorldPos;
+    void Start()
+    {
+        halfWidth = GameManager.GridSystem.GetDimensions().x / 2;
+        halfHeight = GameManager.GridSystem.GetDimensions().y / 2;
+
+        edgeMoveXDampable = new Dampable(edgeMoveDampTime);
+        edgeMoveYDampable = new Dampable(edgeMoveDampTime);
+        scrollMoveDampable = new Dampable(scrollDampTime);
+    }
 
     void Update()
     {
-        // Calculate current camera dimensions
-        Vector2 topRightWorldCoordinates = Camera.main.ScreenToWorldPoint(new Vector3(Screen.width, Screen.height));
-        Vector2 bottomLeftWorldCoordinates = Camera.main.ScreenToWorldPoint(Vector2.zero);
-        Vector2 cameraWorldDimensions = topRightWorldCoordinates - bottomLeftWorldCoordinates;
+        // Get mouse input and update scrollVelocity
+        float scrollVelocity = scrollMoveDampable.currentVelocity + Input.GetAxis("Mouse ScrollWheel") * scrollSensitivity * (invertScrolling ? -1 : 1);
+        scrollMoveDampable.UpdateAndDampen(Input.GetAxis("Mouse ScrollWheel"), scrollVelocity);
```

Comment:
Implementing the Dampable changes as I commented on that class, these lines would change to something like this:

```suggestion
        float scrollAcceleration = Input.GetAxis("Mouse ScrollWheel") * scrollSensitivity * (invertScrolling ? -1 : 1);
        scrollDampable.UpdateSpeed(scrollAcceleration);
```